responsibleai/responsibleai/managers/data_balance_manager.py

```diff
@@ -0,0 +1,433 @@
+        try:
+            (
+                feat_measures_df,
+                dist_measures_df,
+                agg_measures_df,
+            ) = DataBalanceManager._compute_measures(
+                df=df,
+                cols_of_interest=cols_of_interest,
+                target_column=target_column,
+                pos_label=pos_label,
+                backend=backend,
+            )
+
+            (
+                unique_classes,
+                feature_measures,
+            ) = DataBalanceManager._transform_feature_measures(
+                df=feat_measures_df
+            )
+            distribution_measures = (
+                DataBalanceManager._transform_distribution_measures(
+                    df=dist_measures_df
+                )
+            )
+            aggregate_measures = (
+                DataBalanceManager._transform_aggregate_measures(
+                    df=agg_measures_df
+                )
+            )
+
+            return DataBalanceManager._transform_data_balance_measures(
+                unique_classes=unique_classes,
+                feature_measures=feature_measures,
+                distribution_measures=distribution_measures,
+                aggregate_measures=aggregate_measures,
+            )
+        except Exception as e:
+            warnings.warn(f"Failed to get data balance measures due to {e!r}.")
+
+    @staticmethod
+    def _compute_measures(
+        df: Union[pd.DataFrame, Any],
+        cols_of_interest: List[str],
+        target_column: str,
+        pos_label: str,
+        backend: Literal["pandas", "spark"],
+    ) -> Tuple[
+        Optional[pd.DataFrame], Optional[pd.DataFrame], Optional[pd.DataFrame]
+    ]:
+        if backend == "spark":
+            try:
+                return DataBalanceManager._compute_measures_spark(
+                    df=df,
+                    cols_of_interest=cols_of_interest,
+                    target_column=target_column,
+                    pos_label=pos_label,
+                )
+            except Exception as e:
+                warnings.warn(
+                    f"Failed to compute data balance with spark due to {e!r}.",
+                )
+
+        # If spark backend fails or backend == "pandas", use pandas backend
+        try:
+            return DataBalanceManager._compute_measures_pandas(
+                df=df,
+                cols_of_interest=cols_of_interest,
+                target_column=target_column,
+                pos_label=pos_label,
+            )
+        except Exception as e:
+            warnings.warn(
+                f"Failed to compute data balance with pandas due to {e!r}.",
+            )
+
+    @staticmethod
+    def _compute_measures_spark(
+        df: Union[pd.DataFrame, Any],
+        cols_of_interest: List[str],
+        target_column: str,
+        pos_label: str,
+    ) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        from synapse.ml import exploratory as SparkDataBalance
```

Comment:
I think the imports should still go at the top of the file.  You can just add try/catch around them.